nav2_behavior_tree/include/nav2_behavior_tree/plugins/action/spin_action.hpp

```diff
@@ -55,11 +59,28 @@ class SpinAction : public BtActionNode<nav2_msgs::action::Spin>
       {
         BT::InputPort<double>("spin_dist", 1.57, "Spin distance"),
         BT::InputPort<double>("time_allowance", 10.0, "Allowed time for spinning"),
-        BT::InputPort<bool>("is_recovery", true, "True if recovery")
+        BT::InputPort<bool>("is_recovery", true, "True if recovery"),
+        BT::OutputPort<ActionResult::_error_code_type>(
+            "error_code_id", "The spin behavior error code")
```

Comment:
New port to the BT.xml file for use + the behavior tree node xml configuration for Groot